ClosedFunction.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2023 Wanyi He. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Wanyi He
+-/
+import Mathlib.Topology.Sequences
+import Mathlib.Order.LiminfLimsup
+import Mathlib.Topology.Semicontinuous
```

Comment:
This file now contains only/mostly definitions duplicated with the version in `Semicontinuous`. Can we delete this file (potentially by moving more into `Semicontinuous`)?

Reply:
I am very strongly in favor of deleting this file. The terminology "closed function" is super confusing since this has nothing to do with closed maps. There should be a comment in the module docstring of `Mathlib.Topology.Semicontinuous` saying something like: "in some parts of computer science, lower semi-continuous function taking values in (extended) real numbers are called closed functions" so that people who know only this weird terminology can find the relevant file. Maybe "computer science" should be replaced by "applied maths", I'm not sure where this is all coming from.